test/framework/management_cluster.go

```diff
@@ -158,3 +166,87 @@ func WaitForDeploymentsAvailable(ctx context.Context, input WaitForDeploymentsAv
 
 	}, intervals...).Should(BeTrue(), "Deployment %s/%s failed to get status.Available = True condition", input.Deployment.GetNamespace(), input.Deployment.GetName())
 }
+
+// CreateNamespaceInput is the input type for CreateNamespace.
+type CreateNamespaceInput struct {
+	Creator Creator
+	Name    string
+}
+
+// CreateNamespace is used to create a namespace object.
+func CreateNamespace(ctx context.Context, input CreateNamespaceInput, intervals ...interface{}) *corev1.Namespace {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: input.Name,
+		},
+	}
+	By(fmt.Sprintf("Creating namespace %s", input.Name))
+	Eventually(func() error {
+		return input.Creator.Create(context.TODO(), ns)
+	}, intervals...).Should(Succeed())
+
+	return ns
+}
+
+// DeleteNamespaceInput is the input type for DeleteNamespace.
+type DeleteNamespaceInput struct {
+	Deleter Deleter
+	Name    string
+}
+
+// DeleteNamespace is used to delete namespace object.
+func DeleteNamespace(ctx context.Context, input DeleteNamespaceInput, intervals ...interface{}) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: input.Name,
+		},
+	}
+	By(fmt.Sprintf("Deleting namespace %s", input.Name))
+	Eventually(func() error {
+		return input.Deleter.Delete(context.TODO(), ns)
+	}, intervals...).Should(Succeed())
+}
+
+// WatchNamespaceEventsInput is the input type for WatchNamespaceEvents.
+type WatchNamespaceEventsInput struct {
+	ClientSet *kubernetes.Clientset
+	Namespace string
+	LogPath   string
+}
+
+// WatchNamespaceEvents creates a watcher that streams namespace events into a file namespace
```

Comment:
It looks like this is intended to be called from a goroutine and using `context.WithCancel`, could you add that to the godoc and maybe add an example of how it's meant to be used to avoid a consumer believing it is a non-blocking call?

Reply:
Done!